src/tileset/wangset/wang_tile.rs

```diff
@@ -0,0 +1,64 @@
+use std::str::FromStr;
+
+use xml::attribute::OwnedAttribute;
+
+use crate::{
+    error::Error,
+    util::{get_attrs, XmlEventResult},
+    Result, TileId,
+};
+
+/**
+The Wang ID, stored as an array of 8 u32 values.
+*/
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct WangId(pub [u32; 8]);
+
+impl FromStr for WangId {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<WangId, Error> {
+        let mut ret = [0u32; 8];
+        let values: Vec<&str> = s
+            .trim_start_matches('[')
+            .trim_end_matches(']')
+            .split(',')
+            .collect();
+        if values.len() != 8 {
+            return Err(Error::InvalidWangIdEncoding {
+                read_string: s.to_string(),
+            });
+        }
+        for i in 0..8 {
+            ret[i] = values[i].parse::<u32>().unwrap_or(0);
+        }
+
+        Ok(WangId(ret))
+    }
+}
+
+/// Stores a tile reference along with its associated Wang ID.
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct WangTile {
+    /// The Wang ID,
```

Comment:
```suggestion
#[allow(missing_docs)]
```
This doc comment doesn't contribute much so I'd say to get rid of it.